frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -55,13 +86,50 @@ export function DynamicConfiguration () {
         formSubmitting: false as boolean,
     });
 
+    let formErrors = {} as any;
+    const parseResult = formSchema.safeParse(state.form);
+    if (!parseResult.success) {
+        formErrors = issuesToErrors(
+            parseResult.error.issues,
+            state.validateAll === true ? undefined : state.touched
+        );
+    }
+
     useEffect(() => {
         dispatch(listConfigurations({configScope: 'global', numVersions: 5}));
     }, [dispatch]);
 
+    useEffect(() => {
+        dispatch(listEMRApplications());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const appList: any[] = [];
+        for (const application of applicationList) {
```

Comment:
This should probably just be a map: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map